cluster/src/main/java/com/hartwig/pipeline/transfer/SbpFileTransfer.java

```diff
@@ -49,30 +52,24 @@ public void publish(SomaticRunMetadata metadata, SbpRun sbpRun, String sbpBucket
         List<Blob> sourceObjects = find(sourceBucket, metadata.runName());
         List<SourceDestPair> allFiles = new ArrayList<>();
         for (Blob blob : filterStagingBlobs(sourceObjects)) {
-            LOGGER.debug("Syncing object {}", blob.getName());
             if (blob.getMd5() == null) {
                 throw new IllegalStateException(format("Object gs://%s/%s has a null MD5", sourceBucket.getName(), blob.getName()));
             } else {
                 contentTypeCorrection.apply(blob);
-                CloudFile dest = CloudFile.builder()
-                        .provider("s3")
-                        .bucket(sbpBucket)
-                        .path(blob.getName())
-                        .size(blob.getSize())
-                        .md5(blob.getMd5())
-                        .build();
-                CloudFile source = CloudFile.builder()
-                        .provider("gs")
-                        .bucket(sourceBucket.getName())
-                        .path(blob.getName())
-                        .md5(blob.getMd5())
-                        .size(blob.getSize())
-                        .build();
-                allFiles.add(new SourceDestPair(source, dest));
+                allFiles.add(createPair(blob, sbpBucket));
             }
         }
 
+        String manifest = writeManifest(allFiles, metadata.runName());
+        Blob manifestBlob = sourceBucket.get(manifest);
+        if (manifestBlob == null) {
+            throw new RuntimeException("Failed to upload locally-generated manifest");
```

Comment:
maybe can expand a bit on this error message. can imagine it would be a little confusing for an operator (ie what can an operator do in response? Where should they look?)